src/main/java/controllers/ProjectInputController.java

```diff
@@ -0,0 +1,309 @@
+package controllers;
+
+import models.data.Project;
+import models.member.IMember;
+import models.member.Member;
+import models.task.ITask;
+import models.task.Task;
+import repositories.ProjectRepository;
+import util.factories.MemberFactory;
+import util.factories.TaskFactory;
+import util.log.DukeLogger;
+import views.CLIView;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class ProjectInputController implements IController {
+    private Scanner manageProjectInput;
```

Comment:
Consider handling all input at one layer (say CLIView), rather than delegating that responsibility to the controller. Think of a generic 'parser' that carries out high level parsing, and hands over the options/subcommand stuff over to the controller to deal with.

Reply:
Hmm, in that case, will this be a more elegant approach?
`View -> Parser (Helper module) -> Controllers `
With this flow, Parser will take in a Data model of String and return (perhaps) two things to the Controller: the command called and the subcommand/options

The controller relied on the very first word read in to detect the relevant command called by the user. We initially believed that the controller will do some input cleaning before calling the relevant classes for each command. 
Moreover, when modifying things related to a Project, we thought of using a separate Controller to handle commands relevant at that 'instance' of program runtime. If we handled all inputs at one layer, it will result in a cluttered Controller so we decided to abstract it to two Controllers. Is this approach bad? 

Sorry for the trouble!

Reply:
> Hmm, in that case, will this be a more elegant approach?
View -> Parser (Helper module) -> Controllers
With this flow, Parser will take in a Data model of String and return (perhaps) two things to the Controller: the command called and the subcommand/options

Yes, this sounds much cleaner. It's okay to split up your controllers based on similar functionalities. The controllers can be free to do what they want with the 'parsed' object.

Reply:
Understood! Thank you!